src/ucp/tag/offload.c

```diff
@@ -556,9 +557,18 @@ ucs_status_t ucp_tag_offload_sw_rndv(uct_pending_req_t *self)
     rndv_rts_hdr = ucs_alloca(rndv_hdr_len);
     packed_len   = ucp_tag_rndv_rts_pack(rndv_rts_hdr, req);
 
-    return uct_ep_tag_rndv_request(ep->uct_eps[req->send.lane],
-                                   req->send.msg_proto.tag.tag,
-                                   rndv_rts_hdr, packed_len, 0);
+    status = uct_ep_tag_rndv_request(ep->uct_eps[req->send.lane],
+                                     req->send.msg_proto.tag.tag,
+                                     rndv_rts_hdr, packed_len, 0);
+    if (ucs_unlikely(status != UCS_OK)) {
+        if (ucs_likely(status == UCS_ERR_NO_RESOURCE)) {
```

Comment:
No need for checking `ucs_likely(status == UCS_ERR_NO_RESOURCE`, it is already checked in `ucp_rndv_rts_handle_status_from_pending`
can just do `return ucp_rndv_rts_handle_status_from_pending(req, status);`

Reply:
good catch, fixed